src/VolumeRenderSettings.ts

```diff
@@ -0,0 +1,142 @@
+import { Euler, Vector2, Vector3 } from "three";
+import Volume from "./Volume";
+import { Bounds } from "./types";
+
+/**
+ * Marks groups of related settings that may have changed.
+ */
+export enum SettingsFlags {
+  NONE = 0,
+  /** parameters: translation, rotation, scale, currentScale, flipAxes */
+  TRANSFORM = 1,
+  /** parameters: gammaMin, gammaLevel, gammaMax, brightness*/
+  CAMERA = 2,
+  /** parameters: showBoundingBox, boundingBoxColor */
+  BOUNDING_BOX = 4,
+  /** parameters: bounds, zSlice */
+  ROI = 8,
+  /** parameters: maskChannelIndex, maskAlpha */
+  MASK = 16,
+  /** parameters: density, specular, emissive, glossiness */
+  MATERIAL = 32,
+  /** parameters: resolution, useInterpolation, pixelSamplingRate, primaryRayStepSize, secondaryRayStepSize*/
+  RESOLUTION_AND_SAMPLING = 64,
+  /** parameters: isOrtho, orthoScale, viewAxis, visible, maxProjectMode */
+  VIEW = 128,
+  ALL = Number.MAX_SAFE_INTEGER,
+}
+
+export enum Axis {
```

Comment:
Finally added an Axis enum! View3d.ts 468, VolumeDrawable.ts 306 still take strings, however, because I wasn't sure which changes would affect the external API.

Reply:
Also let me know if this belongs somewhere else.

Reply:
Not sure where it should live but it's easy enough to move later.  If we ever promote it to be part of the public API then it will certainly not want to live here.

Reply:
It may be safe to move to the public API now with string values that match what's used in those fields currently ("xy"/"xz"/"yz"/"3d"). (I believe it's technically possible to pass "x"/"y"/"z"? But I don't think it's done.)

Reply:
`setCameraMode` in `View3d.ts` currently takes in a string and doesn't enforce typing ("XY"/"Z"/etc.) on it, so I think for now I'll skip making that change.